cypress/integration/repeated_events.cy.js

```diff
@@ -0,0 +1,207 @@
+import { DIMENSION_ID_ENROLLMENT_DATE } from '../../src/modules/dimensionConstants.js'
+import { ANALYTICS_PROGRAM, TEST_REL_PE_LAST_YEAR } from '../data/index.js'
+import {
+    openDimension,
+    selectEnrollmentProgram,
+    selectEnrollmentProgramDimensions,
+    selectEventProgram,
+} from '../helpers/dimensions.js'
+import { clickMenubarUpdateButton } from '../helpers/menubar.js'
+import { selectRelativePeriod } from '../helpers/period.js'
+import {
+    getTableHeaderCells,
+    expectTableToBeVisible,
+    getTableDataCells,
+} from '../helpers/table.js'
+import { EXTENDED_TIMEOUT } from '../support/util.js'
+
+const getRepeatedEventsTab = () =>
+    cy.getBySel('conditions-modal-content').contains('Repeated events')
+
+const setUpTable = ({ enrollment, dimensionName }) => {
+    selectEnrollmentProgramDimensions({
+        ...enrollment,
+        dimensions: [dimensionName],
+    })
+
+    selectRelativePeriod({
+        label: enrollment[DIMENSION_ID_ENROLLMENT_DATE],
+        period: TEST_REL_PE_LAST_YEAR,
+    })
+
+    cy.getBySel('columns-axis')
+        .findBySel('dimension-menu-button-enrollmentDate')
+        .click()
+    cy.contains('Move to Filter').click()
+
+    cy.getBySel('columns-axis').findBySel('dimension-menu-button-ou').click()
+    cy.contains('Move to Filter').click()
+
+    clickMenubarUpdateButton()
+
+    expectTableToBeVisible()
+
+    cy.getBySelLike('layout-chip').contains(`${dimensionName}: all`)
+}
+
+const setRepetition = ({ dimensionName, recent, oldest }) => {
+    cy.getBySelLike('layout-chip').contains(dimensionName).click()
+    getRepeatedEventsTab().click()
+    cy.getBySel('most-recent-input')
+        .find('input')
+        .type('{backspace}')
+        .type('{moveToEnd}')
+        .type(recent)
+    cy.getBySel('oldest-input')
+        .find('input')
+        .type('{backspace}')
+        .type('{moveToEnd}')
+        .type(oldest)
+    cy.getBySel('conditions-modal').contains('Update').click()
+}
+
+const expectRepetitionToBe = ({ dimensionName, recent, oldest }) => {
+    cy.getBySelLike('layout-chip').contains(dimensionName).click()
+    getRepeatedEventsTab().click()
+    cy.getBySel('most-recent-input')
+        .find('input')
+        .should('be.visible')
+        .and('have.value', recent)
+    cy.getBySel('oldest-input')
+        .find('input')
+        .should('be.visible')
+        .and('have.value', oldest)
+    cy.getBySel('conditions-modal').contains('Hide').click()
+}
+
+const expectHeaderToContainExact = (index, value) =>
```

Comment:
Should this be in the table helper?

Reply:
It could be, but as its only used in this file (so far) and has a pretty specific use case (checking if the header at a specific index has an exact match) I think we could leave it in the file until/if we find another use case for it.